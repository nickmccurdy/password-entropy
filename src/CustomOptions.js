import Options from './Options'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

export default class CustomOptions extends PureComponent {
  static propTypes = {
    onChange: PropTypes.func.isRequired
  }

  state = {
    wordlistLength: 7776
  }

  handleChange = event => this.setState({ wordlistLength: parseInt(event.target.value, 10) })
  possiblePasswords = () => this.state.wordlistLength

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Diceware</h3>
        <label>
          <h4>Wordlist Length</h4>
          <input value={this.state.wordlistLength} onChange={this.handleChange} type="number" min="1" required/>
        </label>
      </Options>
    )
  }
}
