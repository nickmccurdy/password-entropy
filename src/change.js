export default function (wrapper, target) {
  wrapper.simulate('change', { target: { ...wrapper.props(), ...target } })
}
