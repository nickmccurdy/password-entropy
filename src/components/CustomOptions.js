import handleChange from '../handleChange'
import Options from './Options'
import React, { Component } from 'react'

export default class CustomOptions extends Component {
  static propTypes = Options.sharedPropTypes
  static shortName = 'Custom'

  state = {
    possibleItems: 1
  }

  handleChange = handleChange.bind(this)
  possiblePasswords = () => this.state.possibleItems

  render () {
    return (
      <Options possiblePasswords={this.possiblePasswords} {...this.props}>
        <h3>Custom</h3>
        <label>
          <h4>Possible Items</h4>
          <input name="possibleItems" value={this.state.possibleItems} onChange={this.handleChange} type="number" min="1" required/>
        </label>
      </Options>
    )
  }
}
