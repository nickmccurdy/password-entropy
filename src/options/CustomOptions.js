import { setCustom } from '../actions'
import FormGroup from '../ui/FormGroup'
import PropTypes from 'prop-types'
import { path } from 'ramda'
import React, { PureComponent } from 'react'
import { FormControl } from 'react-bootstrap'
import { connect } from 'react-redux'

export class CustomOptions extends PureComponent {
  static propTypes = {
    possibleItems: PropTypes.number.isRequired,
    setCustom: PropTypes.func.isRequired
  }

  render () {
    return (
      <div>
        <h3>Custom</h3>

        <FormGroup id="possibleItems" label="Possible Items" icon="question-circle">
          <FormControl value={this.props.possibleItems} onChange={this.props.setCustom} type="number" min="0" required/>
        </FormGroup>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  possibleItems: path(['options', 'custom'], state)
})

export default connect(mapStateToProps, { setCustom })(CustomOptions)
