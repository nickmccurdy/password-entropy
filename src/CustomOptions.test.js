import change from './change'
import CustomOptions from './CustomOptions'
import { mount, shallow } from 'enzyme'
import React from 'react'

const callback = jest.fn()
const node = (<CustomOptions onChange={callback}/>)

beforeEach(callback.mockReset)

it('renders', () => {
  expect(shallow(node)).toMatchSnapshot()
})

it('calls onChange prop', () => {
  const wrapper = mount(node)
  expect(callback).toHaveBeenCalledWith(1)
  expect(wrapper.state()).toEqual({ possibleItems: 1 })

  callback.mockReset()
  change(wrapper.find({ name: 'possibleItems' }), { value: 2 })
  expect(callback).toHaveBeenCalledWith(2)
  expect(wrapper.state()).toEqual({ possibleItems: 2 })
})
