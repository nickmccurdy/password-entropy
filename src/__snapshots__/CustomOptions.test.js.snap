// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Options
  onChange={[Function]}
  possiblePasswords={[Function]}
>
  <h3>
    Custom
  </h3>
  <label>
    <h4>
      Possible Items
    </h4>
    <input
      min="1"
      name="possibleItems"
      onChange={[Function]}
      required={true}
      type="number"
      value={1}
    />
  </label>
</Options>
`;
